src/core/context/useMessageSaver.tsx

```diff
@@ -16,15 +18,41 @@ export const useMessageSaver = (
     const onChange = async () => {
       const messages = messageHistoryStore.getSnapshot();
       const id = messageHistoryStore.getConversationID();
+      const allConversations: Record<string, ConversationHistory> = JSON.parse(
+        localStorage.getItem('conversations') ?? '{}',
+      );
+
+      // Handle initial system message - save with placeholder
+      if (
+        messages.length === 1 &&
+        messages[0].role === 'system' &&
+        !allConversations[id]
+      ) {
+        // Clean up any existing empty placeholder chats
+        Object.entries(allConversations).forEach(([convId, conv]) => {
+          if (
+            conv.title === PLACEHOLDER_CHAT_TITLE &&
+            conv.conversation.length === 1
+          ) {
+            delete allConversations[convId];
+          }
```

Comment:
Prevents history from being overloaded with empty entries if a user clicks the new chat button repeated or switches between models.